### ISSUE TITLE
Support/filter ground truth datums for SHOW_RADAR intent (e.g., from BenchMap)

### ISSUE THREAD

**Issue by barbeau** [barbeau/gpstest#337]
**Summary:** 

It looks like altitude values that are coming from BenchMap app are in NAVD88, not WGS84:
https://epsg.io/6358

We need to implement filtering and eventually support for these alternate datums via conversions.  A good first step is filtering altitude in non-WGS84 formats to avoid calculating erroneous vertical error values.

One possible way to identify altitude datums is to pass an additional Intent extra value to GPSTest with the key `altitude_epsg` and EPSG value like `6358` for NAVD88.  If the `altitude_epsg` field isn't included in the extra the value is assumed to be WGS84, otherwise the datum specified in the `altitude_epsg` field should be used.

**Steps to reproduce:** 

Share an altitude value from BenchMap

**Expected behavior:** 

Altitude should be in WGS84 if used in error calculations

**Observed behavior:** 

Altitudes are in NAVD88 but would still be used in calculations if provided.

**Device and Android version:** 

N/A

**Comment by gdt** [comment]
That all makes sense.  Keep in mind that "WGS84 altitude" raises the issue of EGM96 vs EGM2008.  I would say  that the right answer is EGM2008 because that is referenced  in the most recent WGS84 specification.

Related to this, I'm seeing very strange geoid offsets, basically -33m when the right answer (EGM2008) for my location is -29m.   I see similar values from an F9P.  So I think there are very coarse geoid models in some GPS receivers, and that's where the -33 is coming from.  GPSTest is presumably able to get the HAE from the android API, but it also seems very likely that the HAE value comes from taking the elevation in NMEA and adjusting it via the reported geoid offset, which I hope is the same value the receiver used to go from HAE to elevation.

It sounds like the right approach is either incorporating proj or reimplementing it, both of which do not seem super easy, but building it with NDK is probably doable.    Perhaps there should be an intent with geojson having a single point, and that already has a standard way to have datum tags.

**Comment by gdt** [comment]
Hey bot: Despite no activity, this is still valid.

**Comment by gdt** [comment]
My quick impression is that Android lives in the WGS84 world and that any app sending intents for locations shouldbe transforming to WGS84 before doing so, rather than extending the API to be datum aware.  If so, that's a big change that would be good to get consensus on from everybody.

**Comment by barbeau** [comment]
@gdt I agree on using WGS84 as the exchange datum - that was the original intent in allowing other apps to share ground truth locations.

**Comment by gdt** [comment]
Hey bot: Despite no activity, this is still valid.

I really should write a bot to post these comments!

**Comment by gdt** [comment]
Hey bot: Despite no activity, this is still valid.

**Comment by barbeau** [comment]
Not stale

**Comment by gdt** [comment]
Hey bot: Despite no activity, this is still valid.

With github's "AI", maybe the bot could read #645!

**Comment by barbeau** [comment]
Not stale
